pkg/storage/chunk/storage/caching_index_client_test.go

```diff
@@ -45,22 +45,35 @@ func TestCachingStorageClientBasic(t *testing.T) {
 	require.NoError(t, err)
 	logger := log.NewNopLogger()
 	cache := cache.NewFifoCache("test", cache.FifoCacheConfig{MaxSizeItems: 10, Validity: 10 * time.Second}, nil, logger)
-	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false)
+	client := newCachingIndexClient(store, cache, 1*time.Second, limits, logger, false, 10, 100)
 	queries := []chunk.IndexQuery{{
 		TableName: "table",
 		HashValue: "baz",
 	}}
 	err = client.QueryPages(ctx, queries, func(_ chunk.IndexQuery, _ chunk.ReadBatch) bool {
 		return true
 	})
+
 	require.NoError(t, err)
+	assert.Eventually(t, func() bool {
+		if asyncClient, ok := client.(IndexAsyncClient); ok {
+			return asyncClient.AsyncQueueLength() == 0
+		}
+		return true
+	}, time.Second, 10*time.Millisecond)
```

Comment:
Not a blocking thing but since this is repeating too often, we could move this to a function `awaitAsyncQueueFlush(*testing.T, *cachingIndexClient)`

Reply:
done,thanks. this is a great suggestion, thanks for the guidance.